pkg/syncer/syncer.go

```diff
@@ -105,6 +108,11 @@ func New(kcpClusterName, pcluster string, fromDiscovery discovery.DiscoveryInter
 		direction:           direction,
 		upstreamClusterName: kcpClusterName,
 		syncerNamespace:     os.Getenv(SyncerNamespaceKey),
+		mutators:            make(mutatorGvrMap),
+	}
+
+	if len(mutators) > 0 {
```

Comment:
Why is it necessary to check the input if we are writing the caller and can assure that what is provided is valid?